src/marshaller/HipieDDL.js

```diff
@@ -1394,14 +1399,17 @@
         this.dashboard.marshaller.commsEvent(this, "request", this.request);
         return new Promise(function (resolve, reject) {
             context.comms.call(context.request).then(function (response) {
-                var delay = 500 - (Date.now() - now);  //  500 is to allow for all "clear" transitions to complete...
-                setTimeout(function () {
-                    context.processResponse(response, request, updates).then(function () {
-                        resolve(response);
-                    });
-                    context.dashboard.marshaller.commsEvent(context, "response", context.request, response);
-                    ++context._loadedCount;
-                }, delay > 0 ? delay : 0);
+                var intervalHandle = setInterval(function () {
+                    if (transactionQueue[0] === myTransactionID && Date.now() - now >= 500) {  //  500 is to allow for all "clear" transitions to complete...
```

Comment:
@jchambers-ln The visualization transitions are guaranteed to finish within the given time frame, what changes on fast/slow hardware is the number of "frames" that can be rendered.

Having said that, I am assuming that all the transitions are at their default duration of 250ms (which I doubled to be sure).  It is not an ideal solution, but is self contained.